File: 2245-destroying-asteroids/destroying-asteroids.cpp
class Solution {
public:
    bool asteroidsDestroyed(int m, vector<int>& arr) {
        sort(arr.begin(),arr.end());
        long long mass=m;
        for(auto &x:arr){
            if(x<=mass)mass+=x;
            else return false;
        }
        return true;
    }
};

File: 2245-destroying-asteroids/destroying-asteroids.java
class Solution {
    public boolean asteroidsDestroyed(int mass, int[] asteroids) {
        Arrays.sort(asteroids);
        long m = mass;
        for (int x : asteroids) {
            if (m >= x)
                m += x;
            else {
                return false;
            }
        }
        return true;
    }
}